streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1678,6 +1678,80 @@ object ZSink extends ZSinkPlatformSpecific {
   final val splitLinesChunk: ZSink[Any, Nothing, Chunk[String], Chunk[String], Chunk[String]] =
     splitLines.contramap[Chunk[String]](_.mkString).mapRemainder(Chunk.single)
 
+  final def splitOn(delimiter: String): ZSink[Any, Nothing, String, String, Chunk[String]] =
+    new SinkPure[Nothing, String, String, Chunk[String]] {
+      type State = SplitOnState
+      case class SplitOnState(
+        // Index into the delimiter
+        i: Int,
+        // Index into the current frame
+        j: Int,
```

Comment:
Nitpicking: I know that the variables are nicely commented but I feel like we could gain readability by giving them slightly more descriptive names, which I think is ok because in this case we are dealing with a specific use case. Feel free to ignore this comment.

P.S. I also wanted to try out multi line comments on GitHub! 😆

Reply:
ditto